ports/esp32s2/mpconfigport.h

```diff
@@ -31,11 +31,14 @@
 #define CIRCUITPY_INTERNAL_NVM_SIZE         (0)
 #define MICROPY_NLR_THUMB                   (0)
 
-#define MICROPY_PY_UJSON                 (1)
-#define MICROPY_USE_INTERNAL_PRINTF      (0)
+#define MICROPY_PY_UJSON                    (1)
+#define MICROPY_USE_INTERNAL_PRINTF         (0)
 
 #include "py/circuitpy_mpconfig.h"
 
+// ESP-IDF leaves this undefined but uses it everywhere.
+// Defining it will reduce chance of undef errors.
+#define CONFIG_IDF_TARGET_ESP32             (0)
```

Comment:
Shouldn't the `-isystem` take care of this?

Reply:
It does for anything that is internal to Circuitpython, but we cannot guarantee that external libraries such as TinyUSB will not include a full path esp-idf file and thus break the `-isystem` fix. @hathach suggested this as an extra layer of protection.

Reply:
I'd rather not add this in case we need it. If submodules use the full IDF path, we can `-isystem` the submodule's IDF path.